### PR TITLE
Documentation: Release Notes for O.5 release

### PR DESCRIPTION
**Pull request by sujitp**
JIRA: MADLIB-700
Release Notes for 0.5 release.

### CODE REVIEW

File: ReleaseNotes.txt [hunk 1]
```diff
@@ -10,6 +10,35 @@ commit history located at https://github.com/madlib/madlib/commits/master.
...
New Features: 
* New Method: Multinomial Logistic Regression
* New Method: Cox Proportional Hazards test
* Added new function for randomly sampling a single element according to weights
* Added new function to simulate EXECUTE … USING
* Added a UDA avg_vector() that computes the average of a set of vectors
* Added support for Linear Chain Conditional Random Fields for NLP
* Added a silhouette function for k-means
* Added a Convex optimization framework
* C++ Abstraction Layer:
```
Comment (iyerr3) [inline ReleaseNotes.txt]: I am wondering why the whole paragraph shows as a single line. Have you added newlines here?

File: ReleaseNotes.txt [hunk 2]
```diff
@@ -10,6 +10,35 @@ commit history located at https://github.com/madlib/madlib/commits/master.
...
* Added support for Linear Chain Conditional Random Fields for NLP
* Added a silhouette function for k-means
* Added a Convex optimization framework
* C++ Abstraction Layer: 
+     - Add Extension to the database abstraction layer to provide support for modular transition states
+
+--------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------
+MADlib v0.4.1
```
Comment (iyerr3) [inline ReleaseNotes.txt]: Please remove the duplicate copy

File: ReleaseNotes.txt [hunk 3]
```diff
@@ -10,6 +10,35 @@ commit history located at https://github.com/madlib/madlib/commits/master.
 Current list of bugs and issues can be found at http://jira.madlib.net.
 
 --------------------------------------------------------------------------------
+MADlib v0.5
+
+Release Date: 2012-Nov-15
+
```
Comment (iyerr3) [inline ReleaseNotes.txt]: I thought we had improved performance in Assoc rules as well. didn't we?

**Comment by geeg** [comment]
It would be good to have the v0.5 release notes in the same format (and modules listed in alphabetical order in each section) as the previous releases, e.g.:

Bug Fixes:
.
.
- k-Means:
  - Improved handling of invalid arguments (MADLIB-359, 361)
    .
    .
    etc.

New Features/Improvements:
- Association Rules:
  - Improved reporting output format for better usability (MADLIB-411)
  - Improved performance
    .
    .
    etc.

Known issues:
- current implementation of Naive Bayes is only suitable for discontinuous attributes (MADLIB-679)
  .
  .
  etc.

**Comment by iyerr3** [comment]
(seconding Grace's suggestion) Considering some of the bugs that are coming out now and since we are moving the fix to next version, I feel it is important to have a 'Know Issues' section that describes the limitations.

**Comment by sujitp** [comment]
Makes sense to me. Will make change and submit another pull request.

**Comment by iyerr3** [comment]
You can just update this branch on your repository (origin) - that will update the pull request automatically.

File: ReleaseNotes.txt [hunk 4]
```diff
@@ -10,6 +10,96 @@ commit history located at https://github.com/madlib/madlib/commits/master.
...
+* Multinomial Logistic Regression: 
+	- Number of independent variables cannot exceed 65535 (MADLIB-665)
+
+* Naive Bayes:
+	- Current implementation of Naive Bayes is only suitable for discontinuous attributes (MADLIB-679)
+	- Inconsistent handling of NULL input values (MADLIB-614)
+	- NULL probabilities given for test set values not seen in training set (MADLIB-523)
+--------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------
```
Comment (iyerr3) [inline ReleaseNotes.txt]: Would be nice to remove the 2nd dashed line.

File: ReleaseNotes.txt [hunk 5]
```diff
@@ -10,6 +10,96 @@ commit history located at https://github.com/madlib/madlib/commits/master.
...
+* SVD Matrix Factorization:
+	- Incremental-gradient SVD algorithm (MADLIB-572)
+
+
+Known issues:
+* Multinomial Logistic Regression: 
+	- Number of independent variables cannot exceed 65535 (MADLIB-665)
+
+* Naive Bayes:
+	- Current implementation of Naive Bayes is only suitable for discontinuous attributes (MADLIB-679)
```
Comment (iyerr3) [inline ReleaseNotes.txt]: I would rather call it categorical attributes (we do accept floating point attributes).

File: ReleaseNotes.txt [hunk 6]
```diff
@@ -10,6 +10,96 @@ commit history located at https://github.com/madlib/madlib/commits/master.
...
+	- Inconsistent handling of NULL input values (MADLIB-614)
```
Comment (iyerr3) [inline ReleaseNotes.txt]: I think MADLIB-614 is not worded right - we don't accept NULL input values at all. The only difference is that an error is not thrown in one case, but the results cannot be trusted in both cases. I would like to change this to NULL values not accepted for attributes.

File: ReleaseNotes.txt [hunk 7]
```diff
@@ -10,6 +10,96 @@ commit history located at https://github.com/madlib/madlib/commits/master.
...
+	- NULL probabilities given for test set values not seen in training set (MADLIB-523)
+--------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------
+MADlib v0.4.1
```
Comment (iyerr3) [inline ReleaseNotes.txt]: I am not sure if this is still the duplicate of the below version.

**Comment by iyerr3** [comment]
looks good, merging this in

**Comment by iyerr3** [comment]
This was merged in https://github.com/madlib/madlib/commit/66dbbed1e6a13b9b65b2a985e176b710164a0a1a

**Pull request closed without merging**